src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -440,7 +440,7 @@ MatterThermostatClusterServerPreAttributeChangedCallback(const app::ConcreteAttr
         }
         auto RequestedSystemMode = static_cast<SystemModeEnum>(*value);
         if (ControlSequenceOfOperation > ControlSequenceOfOperationEnum::kCoolingAndHeatingWithReheat ||
-            RequestedSystemMode > SystemModeEnum::kFanOnly)
+            RequestedSystemMode > SystemModeEnum::kSleep)
```

Comment:
Why are we doing this?  If this is meant to be checking for "valid system mode", then we should be checking that, not hardcoding values that will then break when a new value is added (like just happened).

Reply:
And in particular, the right way to check would be `RequestedSystemMode != SystemModeEnum::kUnknownEnumValue`.

And the check for `ControlSequenceOfOperation` should be fixed too...